scripts/util/math.ts

```diff
@@ -1,40 +1,36 @@
-/**
- * Utility functions for Javascript.
- * Could be ported to C++ and exposed globally in the future.
- */
+export function magnitude(vec: vec2 | vec3): number {
+	return Math.sqrt(sumOfSquares(vec));
+}
 
-/**
- * 2D vector object with x and y member variables
- */
-type Vec2D = { x: number; y: number };
+export function magnitude2D(vec: vec2): number {
```

Comment:
I missed this on first review - sorry! We should be able to get the 2D magnitude of 3D vectors (XY size). The function stays the same, but `vec` should be allowed to be `vec3 | vec2`. CGAZ needs this functionality.